pulp_container/app/utils.py

```diff
@@ -122,3 +123,15 @@ def has_task_completed(dispatched_task):
             task.delete()
             raise Exception(str(error))
     raise Throttled()
+
+
+def get_media_type(content_data, response):
+    """Determine the media type of a manifest either from the JSON data or the response object."""
+    if media_type := content_data.get("mediaType"):
+        return media_type
+    elif media_type := response.headers.get("content-type"):
+        return media_type
+    elif "manifests" in content_data.keys():
```

Comment:
```
elif "manifests" in content_data.keys():
    manifests = content_data.get("manifests")
    if len(manifests):
        if manifests[0].get("media_type") in (docker_v1_schema, docker_v2_schema):
            return docker_manifest_list
        elif manifests[0].get("media_type") in (oci_index, oci_image):
            return oci_index
    return docker_manifest_list
```

Reply:
According to the specs, OCI Index may contain zero manifests: "This REQUIRED property contains a list of [manifests](https://github.com/opencontainers/image-spec/blob/main/manifest.md) for specific platforms. While this property MUST be present, the size of the array MAY be zero.". I find that `if len(manifests)` condition useless.